eZ/Publish/Core/Repository/Helper/RelationProcessor.php

```diff
@@ -70,12 +76,18 @@ public function appendFieldRelations(
 
                 if (isset($destinationIds['locationIds'])) {
                     foreach ($destinationIds['locationIds'] as $locationId) {
-                        if (!isset($locationIdToContentIdMapping[$locationId])) {
-                            $location = $this->persistenceHandler->locationHandler()->load($locationId);
-                            $locationIdToContentIdMapping[$locationId] = $location->contentId;
-                        }
+                        try {
+                            if (!isset($locationIdToContentIdMapping[$locationId])) {
+                                $location = $this->persistenceHandler->locationHandler()->load($locationId);
+                                $locationIdToContentIdMapping[$locationId] = $location->contentId;
+                            }
 
-                        $relations[$relationType][$locationIdToContentIdMapping[$locationId]] = true;
+                            $relations[$relationType][$locationIdToContentIdMapping[$locationId]] = true;
+                        } catch (NotFoundException $e) {
+                            $this->logger->error('Invalid relation: destination location not found', [
+                                'locationId' => $locationId,
+                            ]);
```

Comment:
maybe include `$fieldDefinitionId` also, so it's easier to track Field that refers to missing Location.